### ISSUE TITLE
Crash in pendingFetches array in a serial dispatch queue.

### ISSUE THREAD

**Issue by phamnhatanh1234** [mapbox/mapbox-scenekit#83]
I've seen you guys fix crash on it. Unfortunately, It remains crash when I turn on the AR mode (sometimes). Please check it one more time.

**Comment by xinnai** [comment]
same problem...

**Comment by xinnai** [comment]
I found my code is not the latest ,my version of mapboxscenekit is 0.1.0, which version do you use?

**Comment by phamnhatanh1234** [comment]
I have asked the contributor and he advises me using the another branch: queueAndFetchesCrashFixes.

**Comment by xinnai** [comment]
oh, thank you very much. Did this branch fix your crash? I found it helpfull but I am not sure if crash will happen again because it doesn't appear every time...

**Comment by phamnhatanh1234** [comment]
Yes, this branch fixed my crash. It still crashes but only if you turn on and turn off the AR Map many times (30-40) repeatedly, the crashes now are related to memory storage. 
This email I have received: 
Hi Anh,

Yes I’ve experience this issue quite frequently, fortunately there is a fix out there waiting to be merged in to master. Try using the branch queueAndFetchesCrashFixes. If you need any help with using a specific branch from cocoapods or Carthage I can help you there too!

Have a good one,
Cameron Deardorff
